test/SecTester.Repeater.Tests/RepeaterTests.cs

```diff
@@ -0,0 +1,277 @@
+
+    // act
+    await _sut.Start();
+
+    // assert
+    _timerProvider.Elapsed += Raise.Event<ElapsedEventHandler>(new object(), elapsedEventArgs);
+    await _eventBus.Received(2).Publish(Arg.Is<RepeaterStatusEvent>(x => x.Status == RepeaterStatus.Connected && x.RepeaterId == Id));
+  }
+
+  [Fact]
+  public async Task Start_SetsStatusToRunningJustAfterCall()
+  {
+    // act
+    await _sut.Start();
+
+    // assert
+    _sut.Status.Should().Be(RunningStatus.Running);
+  }
+
+  [Fact]
+  public async Task Start_RepeaterIsStarting_ThrowsError()
+  {
+    // arrange
+    await _sut.Start();
+
+    // act
+    var act = () => _sut.Start();
+
+    // assert
+    await act.Should().ThrowAsync<InvalidOperationException>().WithMessage("Repeater is already active.");
+  }
+
+  [Fact]
+  public async Task Start_AbortedByError_Restarts()
+  {
+    // arrange
+    _eventBus.Execute(Arg.Any<RegisterRepeaterCommand>()).Returns(null, new RegisterRepeaterResult(Version));
+
+    // act
+    var act = () => _sut.Start();
+
+    // assert
+    await act.Should().ThrowAsync<Exception>();
+    await act.Should().NotThrowAsync<Exception>();
+  }
+
+  [Theory]
+  [MemberData(nameof(RegistrationErrors))]
+  public async Task Start_ErrorWhileRegistration_ThrowsError(RegisterRepeaterResult input, string expected)
+  {
+    // arrange
+    _eventBus.Execute(Arg.Any<RegisterRepeaterCommand>()).Returns(input);
+
+    // act
+    var act = () => _sut.Start();
+
+    // assert
+    await act.Should().ThrowAsync<Exception>().WithMessage($"*{expected}*");
+  }
+
+  [Fact]
+  public async Task Start_NewVersionIsAvailable_ShowsWarning()
+  {
+    // arrange
+    var newVersion = new Regex(@"(\d+)").Replace(Version, x => $"{int.Parse(x.Groups[0].Value) + 1}");
+    _eventBus.Execute(Arg.Any<RegisterRepeaterCommand>()).Returns(new RegisterRepeaterResult(newVersion));
+
+    // act
+    await _sut.Start();
+
+    // assert
+    _logger.Received().Log(LogLevel.Warning, Arg.Is<string>(s => s.Contains($"A new Repeater version ({newVersion}) is available")));
+  }
+
+  [Fact]
+  public async Task Stop_RunningRepeater_Stops()
+  {
+    // arrange
+    await _sut.Start();
+
+    // act
+    await _sut.Stop();
+
+    // assert
+    await _eventBus.Received().Publish(Arg.Is<RepeaterStatusEvent>(x => x.Status == RepeaterStatus.Disconnected && x.RepeaterId == Id));
+  }
+
+  [Fact]
+  public async Task Stop_RunningRepeater_EntersIntoOff()
+  {
+    // arrange
+    await _sut.Start();
+
+    // act
+    await _sut.Stop();
+
+    // assert
+    _sut.Status.Should().Be(RunningStatus.Off);
+  }
+
+  [Fact]
+  public async Task Stop_RepeaterIsOff_DoesNothing()
+  {
+    // arrange
+    _eventBus.Execute(Arg.Any<RegisterRepeaterCommand>()).ReturnsNull();
+
+    // act
+    await _sut.Stop();
+
+    // assert
+    _sut.Status.Should().Be(RunningStatus.Off);
+  }
+
+  [Fact]
+  public async Task Stop_RepeaterIsOff_IgnoresSecondCall()
+  {
+    // assert
```

Comment:
AAA typo

Reply:
Is this a new type of typo? Just a typo, not AAA typo 😄